flink-core/src/main/java/org/apache/flink/util/InstantiationUtil.java

```diff
@@ -242,6 +243,10 @@ private static boolean isAnonymousClass(Class clazz) {
 		}
 	}
 
+	private static boolean isAvroSerializer(String name) {
```

Comment:
As discussed offline, I partially agree, and I've added a safeguard for the `streamClassDescriptor`. The local one theoretically can be evolved carefully, but only theoretically. Since the whole idea is to remove Java serialization completely.